### ISSUE TITLE
AuthenticationProtocolMessage.BuildFormPost() produces wrong encoded form input values

### ISSUE THREAD

**Issue by 616b2f** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#755]
Hi,

when i use WsFederationMessage and use BuildFormPost(), the produced form submit can't be readed by wsfed clients.

I think instead of Uri.EscapeDataString() we should use WebUtility.HtmlEncode()

```csharp
public virtual string BuildFormPost()
        {
            StringBuilder strBuilder = new StringBuilder();
            strBuilder.Append("<html><head><title>");
            strBuilder.Append(PostTitle);
            strBuilder.Append("</title></head><body><form method=\"POST\" name=\"hiddenform\" action=\"");
            // instead of strBuilder.Append(Uri.EscapeDataString(IssuerAddress));
            strBuilder.Append(WebUtility.HtmlEncode(IssuerAddress));
            strBuilder.Append("\">");
            foreach (KeyValuePair<string, string> parameter in _parameters)
            {
                strBuilder.Append("<input type=\"hidden\" name=\"");
                // also here
                strBuilder.Append(WebUtility.HtmlEncode(parameter.Key));
                strBuilder.Append("\" value=\"");
                // and here
                strBuilder.Append(WebUtility.HtmlEncode(parameter.Value));
                strBuilder.Append("\" />");
            }

            strBuilder.Append("<noscript><p>");
            strBuilder.Append(ScriptDisabledText);
            strBuilder.Append("</p><input type=\"submit\" value=\"");
            strBuilder.Append(ScriptButtonText);
            strBuilder.Append("\" /></noscript>");
            strBuilder.Append("</form><script language=\"javascript\">window.setTimeout('document.forms[0].submit()', 0);</script></body></html>");
            return strBuilder.ToString();
        }
```

I tested it an with this change it worked well.

**Comment by Tratcher** [comment]
Also encode PostTitle, ScriptDisabledText, ScriptButtonText, etc..

**Comment by brentschmaltz** [comment]
@616b2f  do you want to put together a PR against the dev branch?

**Comment by 616b2f** [comment]
@brentschmaltz I wanna try. I am new two github and open source dev, so feel free to correct me when I do something agains best practice.

Thank you

**Comment by brentschmaltz** [comment]
@616b2f awesome. A working knowledge of Git is a good skill these days. We will walk you through your first commit.
We usually do the following. Create a topic branch off of 'dev', by issuing the git command while in the dev branch: git checkout -b <topic>. Do all the work in that branch, then issue a Pull Request from that branch.

**Comment by brentschmaltz** [comment]
@616b2f @Tratcher  merged PR #757
